src/fabric/src/fabric2_fdb.erl

```diff
@@ -614,6 +616,8 @@ write_doc(#{} = Db0, Doc, NewWinner0, OldWinner, ToUpdate, ToRemove) ->
             end,
             incr_stat(Db, <<"doc_count">>, 1),
             incr_stat(Db, <<"doc_del_count">>, -1);
+        replicate_deleted ->
+            incr_stat(Db, <<"doc_del_count">>, 1);
```

Comment:
Pretty sure you need a second new state because we don't want to increment the deleted count when its deleted/deleted in the state change.

Reply:
A good point. `replicate_deleted_existing` perhaps?